🔀 :: (#156) 디자인 및 버그 수정사항 반영

Hyeonseo4799 opened the pull request:
### 💡 개요
- #156 

### 📃 작업 내용
- 버튼 색상 변경
- 회원가입 화면에서 백스택 버그 수정
- imageUri & imageUris 초기화 버그 수정
- TextField 포커스 버그 수정
- CompositionLocal 대신 entryPoint로 변경

### 💻 구현 화면
Before | After
:--: | :--:
<img src="" width="300" /> | <img src="" width="300" />

### 🎸 기타

Hyeonseo4799 committed:
:lipstick: :: 버튼 색상 변경

Hyeonseo4799 committed:
:bug: :: viewModel state 변경된 상태 유지되는 버그 수정

Hyeonseo4799 committed:
:bug: :: imageUri 및 imageUris 초기화 버그 수정

Hyeonseo4799 committed:
Merge branch 'feature/157_publishing_to_create_screen' into feature/156_design_and_bug_fix

Hyeonseo4799 committed:
:bug: :: TextField 포커스 문제 수정

Hyeonseo4799 committed:
:recycle: :: CompositionLocal 대신 entryPoint 사용

Cjsghkd commented on feature/create/src/main/java/com/idiotfrogs/create/CreateScreen.kt:
```diff
@@ -45,6 +49,12 @@ fun CreateScreen(modifier: Modifier = Modifier) {
     val enabled = titleTextFieldState.text.isNotEmpty() && contentTextFieldState.text.isNotEmpty()
 
+    val titleInteractionSource = remember { MutableInteractionSource() }
+    val titleIsFocused by titleInteractionSource.collectIsFocusedAsState()
```
해당 반복 되는 로직도 확장함수로 두면 좋을 것 같습니다.

Hyeonseo4799 replied:
아래 커밋에서 수정했습니다
https://github.com/IdiotFrogs/MemorySeal-Android/pull/159/commits/2cd9cd71f40289a31f3ccd7de35cc9177313f258

Hyeonseo4799 committed:
:sparkles: :: interactionSource 및 isFocus 확장 함수화

Hyeonseo4799 committed:
:sparkles: :: rememberFocusState 적용

Cjsghkd commented on core/designsystem/src/main/java/com/idiotfrogs/designsystem/component/MSTextField.kt:
```diff
@@ -108,9 +110,13 @@ fun MSTextField(
 @Composable
 private fun MSTextFieldPreview() {
     val textFieldState = rememberTextFieldState()
+    val interactionSource = remember { MutableInteractionSource() }
+    val isFocused by interactionSource.collectIsFocusedAsState()
```
해당 부분 Preview 들도 확장함수 적용 부탁드립니다.

Cjsghkd commented on feature/auth/src/main/java/com/idiotfrogs/auth/signup/SignUpScreen.kt:
```diff
@@ -77,6 +79,9 @@ fun SignUpScreen(
     val scrollState = rememberScrollState()
     val (imageUri, launchImagePicker) = rememberPickerState()
 
+    val interactionSource = remember { MutableInteractionSource() }
+    val isFocused by interactionSource.collectIsFocusedAsState()
+
```
이 코드도 마찬가지로 적용 부탁드립낟.

Cjsghkd commented on feature/home/src/main/java/com/idiotfrogs/home/component/HomeJoinContainer.kt:
```diff
@@ -38,6 +41,8 @@ fun BoxScope.HomeJoinContainer(
     onCancel: () -> Unit,
     onJoin: () -> Unit,
 ) {
+    val interactionSource = remember { MutableInteractionSource() }
+    val isFocused by interactionSource.collectIsFocusedAsState()
```
해당 코드도 적용 부탁드립니다.

Hyeonseo4799 committed:
:recycle: :: PR 리뷰 반영

- isFocused 및 interactionSource 디폴트를 확장함수로 받도록 변경
- textFieldState 디폴트를 확장함수로 받도록 변경

Hyeonseo4799 committed:
:sparkles: :: 화면 간 네비게이션 처리

Cjsghkd committed:
Merge remote-tracking branch 'refs/remotes/origin/feature/156_design_and_bug_fix' into feature/160_screen_navigation

# Conflicts:
#	app/src/main/java/com/idiotfrogs/memoryseal/MainActivity.kt

Cjsghkd committed:
:recycle: 네비게이션 로직 수정

Cjsghkd committed:
:recycle: fetchInitUi Init 상태 초기화 추가 및 불 필요한 DisposableEffect 제거

Cjsghkd committed:
:sparkles: core:util 모듈 생성

Cjsghkd committed:
:sparkles: core:util 모듈 의존성 주입 및 다른 core 모듈 의존성도 주입

Cjsghkd committed:
:sparkles: 공용 UiState 생성 및 적용

Cjsghkd committed:
Merge pull request #161 from IdiotFrogs/feature/160_screen_navigation

🔀 :: (#160) 화면 간 네비게이션 처리
